src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java

```diff
@@ -0,0 +1,239 @@
+				project.getProperties().put(CCSKAFKAVERSION, highestCCSVersion.toString());
+			}
+		} catch (VersionRangeResolutionException e) {
+			throw new MojoExecutionException("", e);
+		}
+	}
+
+	private Version fetchHighestKafkaVersion(String kafkaType, VersionRangeResult result, String constraintText)
+			throws MojoExecutionException, MojoFailureException {
+
+		// Workaround for https://issues.apache.org/jira/browse/MNG-3092
+		try {
+			getLog().debug(kafkaType + " version fetched " + result.getVersions());
+			if (!includeSnapshots) {
+				result.setVersions(removeSnapshots(result.getVersions()));
+			}
+			if (kafkaType.equals(CE)) {
+				result.setVersions(includeCE(result.getVersions()));
+				// if ce-kafka can not be fetched
+				if (result.getVersions().isEmpty()) {
+					getLog().info(CE + "can not be fetched");
+				}
+			} else if (kafkaType.equals(CCS)) {
+				result.setVersions(includeCCS(result.getVersions()));
+			} else {
+				throw new MojoFailureException("kafka type " + kafkaType + "wrong or not supported");
+			}
+
+			getLog().debug(kafkaType + " Constraint: " + result.getVersionConstraint());
+			getLog().info(kafkaType + " Versions in range: " + result.getVersions());
+
+			Version highestVersion = result.getHighestVersion();
+
+			if (highestVersion == null) {
+				throw new MojoFailureException(
+						"No matching " + kafkaType + " version found for constraint: '" + constraintText + "'.");
+			}
+
+			return highestVersion;
+		} catch (MojoFailureException e) {
+			throw new MojoExecutionException("", e);
+		}
+	}
+
+	public Version fetchHighestKafkaVersionPublic(String kafkaType, VersionRangeResult result, String constraintText)
+			throws MojoExecutionException, MojoFailureException {
+		return fetchHighestKafkaVersion(kafkaType, result, constraintText);
+	}
+
+	private List<Version> removeSnapshots(List<Version> versions) {
+		ArrayList<Version> result = new ArrayList<>();
+		for (Version version : versions) {
+			if (!isSnapshot(version.toString())) {
+				result.add(version);
+			}
+		}
+		return result;
+	}
+
+	private List<Version> includeCE(List<Version> versions) {
+		ArrayList<Version> result = new ArrayList<>();
+		for (Version version : versions) {
+			if (isCE(version.toString())) {
+				result.add(version);
+			}
+		}
+		return result;
+	}
+
+	private List<Version> includeCCS(List<Version> versions) {
+		ArrayList<Version> result = new ArrayList<>();
+		for (Version version : versions) {
+			if (isCCS(version.toString())) {
+				result.add(version);
+			}
+		}
+		return result;
+	}
+
+	/**
+	 * Copied from org.eclipse.aether.artifact.AbstractArtifact.isSnapshot(String).
+	 */
+	private static boolean isSnapshot(String version) {
+		return version.endsWith(SNAPSHOT) || SNAPSHOT_TIMESTAMP.matcher(version).matches();
+	}
+
+	private static boolean isCCS(String version) {
+		return version.endsWith(CCSVersionString);
```

Comment:
Should this be endsWith or matching a regex? Seems like endsWith wouldn't work for SNAPSHOTs.

Reply:
There is another logic that remove SNAPSHOT first.